programs/fileio.c

```diff
@@ -1689,12 +1684,28 @@ int FIO_compressMultipleFilenames(FIO_prefs_t* const prefs,
             ress.dstFile = NULL;
         }
     } else {
-        unsigned u;
+        unsigned int u=0;
+        if (outMirroredRootDirName)
+            UTIL_mirrorSourceFilesDirectories(inFileNamesTable, nbFiles, outMirroredRootDirName);
+
         for (u=0; u<nbFiles; u++) {
             const char* const srcFileName = inFileNamesTable[u];
-            const char* const dstFileName = FIO_determineCompressedName(srcFileName, outDirName, suffix);  /* cannot fail */
+            const char* dstFileName = NULL;
+            if (outMirroredRootDirName) {
+                char* outMirroredDirName = UTIL_createMirroredDestDirName(srcFileName, outMirroredRootDirName);
+                if (outMirroredDirName) {
+                    dstFileName = FIO_determineCompressedName(srcFileName, outMirroredDirName, suffix);
+                    free(outMirroredDirName);
+                } else {
+                    continue;
```

Comment:
Reaching that point is an error (basically `outMirroredDirName` allocation failed).
It should at least trigger some error signal (`error = 1;`).

Reply:
right... 

I have a test case for this in the playTests.sh
```
    println "test: compress input dir will be ignored if it has '..'"
    zstd  -r tmpInputTestDir/we/must/../must --output-dir-mirror non-exist
    test ! -d non-exist
```
if I add that `error=1`, the zstd will exit with error signal and break the test suite. 

Is there a way to handle this in test script?

Reply:
Ah, that's different then.
Indeed, it matters to define a behavior.

First, is it certain that, on reaching that branch of the code, the reason necessarily is that a `..` is present in the filename ?
If yes, this should be documented.

I hear that `UTIL_createMirroredDestDirName()` can fail, but when it does, the failure reason cannot be an allocation error.
That's unusual for a function which returns a `NULL` pointer. But well, ok, unusual isn't impossible. Just, at least, it must be clearly documented.

Then is that a good idea to fail silently without sending an error signal ?
I don't think so.
For example, `zstd badfile goodfile` will fail if `goodfile` exists but `badfile` doesn't exist. 
It will nonetheless continue beyond `badfile`, and generate `goodfile.zst`.

I see no difference with the case described above, where compression of a file is skipped because of a problem, but the rest of the list proceed nonetheless. It should still return an error code.

As for the test itself, testing that an error is raised is not difficult.
We already have multiple tests which check that a given scenario is supposed to fail.
See for example : https://github.com/facebook/zstd/blob/v1.4.5/tests/playTests.sh#L235 .

Reply:
`That's unusual for a function which returns a NULL pointer. But well, ok, unusual isn't impossible. Just, at least, it must be clearly documented.`

Yeah, I overload the return meaning a bit. I changed the name from `outMirroredDirName` to `validMirroredDirName`, and add a comment to `UTIL_createMirroreDestDirName`, hopefully it reduce the confusion a bit. here. 

I decided not to throw error for this case (same as 'tar' case) but just do printf to stderr.

Reply:
That's better, but I still believe it's more in line with other functionalities of `zstd` to issue an error status in this case.
I understand that `tar` wouldn't. But for such a case, we do have a reference in `zstd` directly, which takes precedence.
(and not just `zstd` by they way, `gzip` and `xz` are also excellent references, and they all issue an error code when one element in the list of objects to compress or decompress is not electable for any sort of condition (file does not exist, or file is a directory, for example)).

Reply:
Agree, I will do so.